be/src/olap/olap_server.cpp

```diff
@@ -390,7 +379,7 @@ void StorageEngine::_compaction_tasks_producer_callback() {
         } else {
             sleep(config::check_auto_compaction_interval_seconds);
         }
-    }
+    } while (!_stop_background_threads_latch.wait_for(MonoDelta::FromMilliseconds(interval)));
```

Comment:
Need to call `THREAD_JOIN(_compaction_tasks_producer_thread);` in StorageEngine::stop(), otherwise it's still unsafe as described in https://github.com/apache/incubator-doris/issues/5213

Reply:
OK.

Reply:
@weizuo93 @acelyc111 
Is _compaction_thread_pool also need call shutdown?
It seems _compaction_thread_pool still depend on 'this->_permit_limiter'

Reply:
@stdpain 
 `_compaction_thread_pool` has been shutdown in `~StorageEngine()`.